wasp-lang/wasp
# Implement the project install step

Fixes #1597.

Tasks:
- [x] Add an extra install step during `wasp start`
- [x] Decide what to do with the `dependencies` field in `.wasp` files - I decided to remove the field and only keep a single source of truth, the `_package.json_ file.
- [x] Update the docs.
- [x] Fix wasp not installing npm deps that are in package.json but are not installed. -> fixed in https://github.com/wasp-lang/wasp/pull/1652 .
- [ ] TODO: https://github.com/wasp-lang/wasp/pull/1602/files#diff-aa79f536fd3bb9203f9d976c63594a52a9f3533759e2b754e2a6baec6403a7dbR95-R96 -> this is the one that the only unresolved comment is about.


How to run and test stuff:
- Go to `waspc/examples/todo-typescript` and work inside that project.
- Running `./cleanstart` will clean everything, fix known issues (e.g., duplicate react) and start the app with all the necessary migrations applied.
- The rest of the commands (e.g,. `cabal run wasp-cli start`, `cabal run wasp-cli clean`) will do what you expect.
- After each change, either let the application pick it up, rerun `cabal run wasp-cli start`, or run `./cleanstart` (depending on what you're changing).
- You can run `./fix` to fix the double react installation issue (but often won't need to, as `./cleanstart` takes care of it). 
- Please don't commit the migrations or `package-lock.json`.

> [!IMPORTANT]  
> **PR 1603** (`filip-wasp-reset`) branches out of **PR 1604** (`filip-restructuring-prisma`), which branches out of **PR 1602** (`filip-project-install-step`).
> The PR numbers are unfortunately out of order.
> Make sure you keep the branches up to date with their bases as you change stuff.
